src/gtc/passes/oir_optimizations/utils.py

```diff
@@ -46,14 +47,23 @@ def visit_FieldAccess(
         is_write: bool,
         **kwargs: Any,
     ) -> None:
-        if isinstance(node.offset.k, int):
-            accesses.append(
-                Access(
-                    field=node.name,
-                    offset=(node.offset.i, node.offset.j, node.offset.k),
-                    is_write=is_write,
-                )
+        self.visit(node.offset, accesses=accesses, is_write=is_write, **kwargs)
```

Comment:
If this is the only thing left in the visitor method, I'm not sure it's needed any more.